tests/python_package_test/test_engine.py

```diff
@@ -1545,6 +1563,39 @@ def test_refit():
     assert err_pred > new_err_pred
 
 
+def test_refit_kwargs_for_predict(artifacts_for_refit_kwargs):
+    # check refit accepts kwargs_for_predict
+    X, label, booster = artifacts_for_refit_kwargs
+    kwargs_for_dataset = {
+        "weight": [1.0, 0.0, 1.0],
+        "reference": None,
+        "group": None,
+        "init_score": None,
+        "feature_name": "auto",
+        "categorical_feature": "auto",
+        "free_raw_data": True
+    }
+    booster_refit = booster.refit(
+        X, label, kwargs_for_dataset=kwargs_for_dataset
+    )
+    pred = booster_refit.predict(X)
+    assert pred.shape == (3, )
```

Comment:
I think that this assertion would be true whether or not the `weight` passed above was used, wouldn't it? If I'm right about that, then this is not effectively testing the changes in this PR.

Please test that the results you get are different based on the weights provided.

You could use the following as an example:

https://github.com/microsoft/LightGBM/blob/8a34b1af2d5ae734a446f97ee984508a55ecb2d4/tests/python_package_test/test_engine.py#L528-L537

Reply:
@jameslamb just to double check, the test should prove that a weighted sample gives different metric value than a non-weighted, but not necessarily a better score because weights are random `weight=np.abs(np.random.normal(size=y.shape))`, is that right?

Reply:
yes, as long as choosing the weights is the only source of randomness. Just something that gives us confidence that weights are actually being used inside this method's code.